.github/pull_request_template.md

```diff
@@ -0,0 +1,46 @@
+<!-- Thank you for submitting a pull request! Find more information in our
+development guide at 
+https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md
+and contact us on #dev in slack. -->
+## Description
+<!-- Elaborate beyond the title of the PR as necessary to help the reviewers.-->
+
+## Related issues and pull requests
+<!-- List any issue that is fixed or related, such as documentation PR:
+
+* Fixes #issuenumber
+* Related documentation in #issuenumber
+-->
+
+## Documentation
+<!-- 
+To fill in by maintainer with input from contributor:
+Remove x to confirm that no documentation and/or no release notes entry is
+needed, or provide relevant information.-->
+
+- [x] User facing documentation needed
+- [x] Release notes entry needed
+
+## Release notes
+<!--
+To fill in by maintainer with input from contributor:
+  Add all needed release notes entries with section title, wording, and link
+  to related issue or pull request. Sections include General, Security,
+  JDBC driver, CLI, SPI, and connectors. If PR affects multiple sections,
+  especially also multiple connectors, list them all with the desired wording.
+  Use markdown syntax:
+
+## General
+
+* Improver performance of so and so. ({issue}`1234`)
+* Fix incorrect results .... type. ({issue}`5678`)
+
+## Hive connector
+
+* Add support for super cool feature. ({issue}`9123`)
+
+## Iceberg connector
+
+* Add support for super cool feature. ({issue}`91234`)
```

Comment:
The PR template doesn't (and shouldn't) contain RN template. 

Also, we currently collect RNs under a RN issue.
I don't think anything changed, and there even was no Maintainers' dicussion advocating for such a change.

Reply:
> I don't think anything changed, and there even was no Maintainers' dicussion advocating for such a change.

We've had some ad-hoc discussions and hence the PR. We've talked about possibly doing something like this many months ago (how we might streamline the release note collection process), and @mosabua now gracefully offered to help try some ideas and drive this forward.

Reply:
Collecting RNs under PR doesn't scale well IMO with external contributors - now non-PR related maintainanence discussion ends up happening during PR review adding noise.
Also it's easy to end up with outdated release notes since the actual changes being merged is not final until the PR is actually merged.

I'd propose to exclude the RN template for now - we can iterate on this over time.

Reply:
I strongly disagree. A proper RN entry and documentation should be part of the work of getting a PR ready for merge and everything should be done by the time something is merged. This work can and should be done by the maintainer helping the contributor to get to merge. And it has to explicitly communicate if a RN or docs are needed for each PR. 

> > Here is a live PR that used the template already manually - #10898
> 
> i read there
> 
> > * Update JDBC driver to 10.2.0 ({issue}`10898`)
> 
> This is a good example why we shouldn't ask contributors for release notes.
> 
> in that PR, a contributor didn't now the release notes need to present the change from user perspective. Version upgrade of an underlying library hardly translates to user visible impact, unless user does their job and checks why we could have done the upgrade.

If you actually looked at the PR in detail you would see that this is WIP and only a draft proposal. Part of getting the PR ready is to figure out the correct release notes entry and the PR says as much. There are most likely user facing changes and they are in the linked upstream release notes. By the time this PR is ready it should include a release notes entry that is ready to be used nearly unmodified. Since this is WIP that is of course not the case yet..

Reply:
>  Also it's easy to end up with outdated release notes since the actual changes being merged is not final until the PR is actually merged.

And there is no mechanism to indicate that the RN is no longer a draft or that someone forgot to update it.

Reply:
Sure .. everything in the PR is valid until merge. Then the release notes text is absorbed into the release notes PR .. and iteration continues there if necessary.